src/olympia/activity/utils.py

```diff
@@ -155,30 +155,35 @@ def add_email_to_activity_log(parser):
 
     version = token.version
     user = token.user
-    if token.is_valid():
-        log_type = action_from_user(user, version)
-
-        if log_type:
-            note = log_and_notify(log_type, parser.reply, user, version)
-            log.info('A new note has been created (from %s using tokenid %s).'
-                     % (user.id, uuid))
-            token.increment_use()
-            return note
+    if user.banned is None:
```

Comment:
You should follow the other error conditions and raise an exception (and test that it's raised in the unit test). That ensures all the error conditions go through similar code paths.

Reply:
703df2c555